### PR TITLE
KEYCLOAK-18906 Wait for the test app to start

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

File: scripts/run-tests.sh [hunk 1]
```diff
@@ -97,6 +97,15 @@ run_tests app-springboot -Pspring-boot
 # service-nodejs tests
 npm -C service-nodejs install
 npm -C service-nodejs start >/dev/null&
+# Wait for port 3000 to open for at most 30 seconds
+{
+    I=0
+    while ! echo -n > /dev/tcp/localhost/3000 && [[ $I -lt 60 ]]; do
```
Comment (stianst) [inline scripts/run-tests.sh:103]: This just checks if socket is open, is there a window until it's able to process requests? Might it be better to use curl or wget to check it's ready returns a valid HTTP response?

**Review by stianst — APPROVED** [review]
LGTM, added a comment with regards to perhaps there's a small window, but that is probably overthinking it, so happy either way

File: scripts/run-tests.sh (hunk 1 above)
Comment (hmlnarik) [reply scripts/run-tests.sh]: Yup, that's correct. Even if the window would be negligible, curl would be a bit more reliable, so I'm pushing the new version with curl

**Review by stianst — APPROVED** [review]

File: scripts/run-tests.sh [hunk 2]
```diff
@@ -97,6 +97,15 @@ run_tests app-springboot -Pspring-boot
...
 npm -C service-nodejs start >/dev/null&
+# Wait for port 3000 to open for at most 30 seconds
+{
+    I=0
+    while ! curl -sfN -o /dev/null http://localhost:3000/service/public && [[ $I -lt 60 ]]; do
+         sleep 0.5
+         echo -n .
+         I=$[$I + 1]
+    done
+} 2>/dev/null
```
Comment (stianst) [inline scripts/run-tests.sh:108]: Curl supports retry itself, so you can do:
```suggestion
# Wait for port 3000 to open for at most 30 seconds
curl --retry 60 --retry-delay 1 --retry-all-errors -sfN -o /dev/null http://localhost:3000/service/public
```

**Pull request merged**